### PR TITLE
feat: Add Event tabs in pipeline run record

### PR DESCRIPTION
**Pull request by harrisonliu5**
Signed-off-by: harrisonliu5 <harrisonliu@kubesphere.io>

### What type of PR is this?

/kind feature

### Which issue(s) this PR fixes:

Fixes #https://github.com/kubesphere/console/issues/3003

### Special notes for reviewers:

image

### Does this PR introduced a user-facing change?

```release-note
feat: Add Event tabs in pipeline run record
```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

File: src/pages/devops/containers/Pipelines/Detail/Events/index.jsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
+/*
+ * This file is part of KubeSphere Console.
+ * Copyright (C) 2019 The KubeSphere Console Authors.
```
Comment (LinuxSuRen) [inline src/pages/devops/containers/Pipelines/Detail/Events/index.jsx:3]: `2019` or `2022`?

File: src/pages/devops/containers/Pipelines/Detail/Events/index.jsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
+/*
+ * This file is part of KubeSphere Console.
```
Comment (LinuxSuRen) [inline src/pages/devops/containers/Pipelines/Detail/Events/index.jsx:2]: why this is part of?

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]

File: src/pages/devops/containers/Pipelines/Detail/Events/index.jsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  fetchData() {
+    const { _originData = {} } = this.store.runDetail
+
+    const { devops, runName } = this.props.match.params
+
+    const fields = {
+      'involvedObject.name': runName,
+      'involvedObject.namespace': devops,
+      'involvedObject.kind': _originData.kind,
+      'involvedObject.uid': get(_originData, 'metadata.uid', ''),
```
Comment (JohnNiang) [inline src/pages/devops/containers/Pipelines/Detail/Events/index.jsx:62]:
```suggestion
      'involvedObject.name': _originData.metadata.name,
      'involvedObject.namespace': _originData.metadata.namespace,
      'involvedObject.kind': _originData.kind,
      'involvedObject.uid': _originData.metadata.uid,
```

**Review by JohnNiang — CHANGES_REQUESTED** [review]
All code are good to me, but those lines. I have verified the changes work as same as before.

File: src/pages/devops/containers/Pipelines/Detail/Events/index.jsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    this.eventStore.fetchList({
+      namespace: devops,
```
Comment (JohnNiang) [inline src/pages/devops/containers/Pipelines/Detail/Events/index.jsx:66]:
```suggestion
      namespace: _originData.metadata.namespace,
```

File: src/pages/devops/containers/Pipelines/Detail/Events/index.jsx (hunk 1 above)
Comment (harrisonliu5) [reply src/pages/devops/containers/Pipelines/Detail/Events/index.jsx:3]: This is a general description

File: src/pages/devops/containers/Pipelines/Detail/Events/index.jsx (hunk 2 above)
Comment (harrisonliu5) [reply src/pages/devops/containers/Pipelines/Detail/Events/index.jsx:2]: This is a general description

File: src/pages/devops/containers/Pipelines/Detail/Events/index.jsx (hunk 3 above)
Comment (harrisonliu5) [reply src/pages/devops/containers/Pipelines/Detail/Events/index.jsx:62]: maybe the `originData` is empty, so need the `get` function to init the data. @JohnNiang

**Comment by harrisonliu5** [comment]
@JohnNiang @LinuxSuRen  please review again, thaks

File: src/pages/devops/containers/Pipelines/Detail/Events/index.jsx (hunk 1 above)
Comment (LinuxSuRen) [reply src/pages/devops/containers/Pipelines/Detail/Events/index.jsx:3]: `The year in the copyright notice refers to the year of first publication`

See also https://opensource.stackexchange.com/questions/6389/how-do-the-years-specified-in-a-copyright-statement-work

but I don't block anyone to merge this PR.

**Comment by LinuxSuRen** [comment]
In case of anyone might want to use image to test this PR, I post the image following:

```
surenpi/ks-console:latest@sha256:1066563c09d6c75b17b0e848c9c9c76e0eabe23c0cf02a1b1f7af29f37f9267e
```

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]
It does not work well if I directly visit this page instead of switching to it.

image

**Review by JohnNiang — APPROVED** [review]
I can confirm that the issue mentioned https://github.com/kubesphere/console/pull/3012#pullrequestreview-921087873 has been fixed by the latest commits.

image

/lgtm

**Pull request merged**